metapool/count.py

```diff
@@ -81,26 +83,45 @@ def _parsefier(run_dir, sample_sheet, subdir, suffix, name, funk):
     """
     out = []
 
-    for project, lane in {(s.Sample_Project, s.Lane) for s in sample_sheet}:
+    is_sample_sheet = True if isinstance(metadata,
+                                         metapool.KLSampleSheet) else False
+
+    if is_sample_sheet:
+        projects = {(s.Sample_Project, s.Lane) for s in metadata}
+    else:
+        projects = metadata[["Project_name", "lane"]]
+        projects = set(projects.itertuples(index=False, name=None))
+
+    for project, lane in projects:
         lane = lane.zfill(3)
 
-        # log files are named after the sequence files themselves, specifically
-        # the forward sequences, so we just make sure we match the right lane,
-        # the forward file and the suffix. The suffix is something like ".log"
-        # or "*.json" if you expect to see other characters before the
-        # extension
+        # log files are named after the sequence files themselves,
+        # specifically the forward sequences, so we just make sure we
+        # match the right lane, the forward file and the suffix. The
+        # suffix is something like ".log" or "*.json" if you expect to see
+        # other characters before the extension.
         logs = glob.glob(os.path.join(run_dir, project, subdir,
                                       f'*_L{lane}_R1_001' + suffix))
 
         for log in logs:
             out.append([*_extract_name_and_lane(os.path.basename(log)),
                         project, log])
 
-    out = pd.DataFrame(columns=['Sample_ID', 'Lane', 'Sample_Project', 'path'],
-                       data=out)
+    out = pd.DataFrame(
+        # TODO: Should this be Sample_ID for mapping-files? It's using the
+        #  mapping-file's sample-name? Perhaps this column should be renamed
+        #  or the data-frame should use the correct column. Note 'lane' is
+        #  now 'Lane' and 'Project_name' is 'Sample_Project'.
+        columns=['Sample_ID', 'Lane', 'Sample_Project', 'path'],
+        data=out)
 
+    # note that expected uses our generated Sample_ID for amplicon and not
+    # sample_name as it should.
     found = set(out['Sample_ID'])
-    expected = {s.Sample_ID for s in sample_sheet}
+    if is_sample_sheet:
+        expected = {s.Sample_ID for s in metadata}
+    else:
+        expected = set(metadata["Sample_ID"].values.tolist())
```

Comment:
This can actually be added in the first if/else, correct? I think there is no need for a separate block.

Reply:
ty for the catch!